src/lightning_app/core/work.py

```diff
@@ -119,7 +119,16 @@ def __init__(
                 " in the next version. Use `cache_calls` instead."
             )
         self._cache_calls = run_once if run_once is not None else cache_calls
-        self._state = {"_host", "_port", "_url", "_future_url", "_internal_ip", "_restarting", "_cloud_compute"}
+        self._state = {
+            "_host",
+            "_port",
+            "_url",
+            "_future_url",
+            "_internal_ip",
+            "_restarting",
+            "_cloud_compute",
+            "_display_name",
```

Comment:
@tchaton Is the display name meant to be changeable? Will the UI update the name when it changes in the state?

Reply:
You are right ! Added an exception if the work has already started.